File: ScaleformUI_V8/src/elements/controls.ts
export class Controls {
    public static NecessaryControlsKeyboard = [
        201, 195, 196, 187, 188, 189, 190, 202, 217, 242, 241, 239, 240, 237, 238,
        31, 30, 21, 22, 23, 75, 71, 72, 59, 89, 9, 8, 90, 76,
    ]
    public static NecessaryControlsGamePad = [
        201, 195, 196, 187, 188, 189, 190, 202, 217, 242, 241, 239, 240, 237, 238,
        31, 30, 21, 22, 23, 75, 71, 72, 59, 89, 9, 8, 90, 76, 2, 1, 25, 24, 71, 72,
        59, 31, 30, 75,
    ]

    public static toggleAll(toggle: boolean) {
        if (toggle) {
            EnableAllControlActions(0)
            EnableAllControlActions(1)
            EnableAllControlActions(2)
        } else {
            DisableAllControlActions(2);
            (IsUsingKeyboard(2) ? this.NecessaryControlsKeyboard : this.NecessaryControlsGamePad)
                .forEach(ctrl => EnableControlAction(0, ctrl, true))
        }
    }
}

File: ScaleformUI_Lua/src/Elements/Controls.lua
Controls = setmetatable({
    NecessaryControlsKeyboard = {
        201, -- FrontendAccept
        195, -- FrontendAxisX
        196, -- FrontendAxisY
        187, -- FrontendDown
        188, -- FrontendUp
        189, -- FrontendLeft
        190, -- FrontendRight
        202, -- FrontendCancel
        217, -- FrontendSelect
        242, -- CursorScrollDown
        241, -- CursorScrollUp
        239, -- CursorX
        240, -- CursorY
        237, -- CursorAccept
        238, -- CursorCancel
        31,  -- MoveUpDown
        30,  -- MoveLeftRight
        21,  -- Sprint
        22,  -- Jump
        23,  -- Enter
        75,  -- VehicleExit
        71,  -- VehicleAccelerate
        72,  -- VehicleBrake
        59,  -- VehicleMoveLeftRight
        89,  -- VehicleFlyYawLeft
        9,   -- FlyLeftRight
        8,   -- FlyUpDown
        90,  -- VehicleFlyYawRight
        76   -- VehicleHandbrake
    },

    NecessaryControlsGamePad = {
        201, -- FrontendAccept
        195, -- FrontendAxisX
        196, -- FrontendAxisY
        187, -- FrontendDown
        188, -- FrontendUp
        189, -- FrontendLeft
        190, -- FrontendRight
        202, -- FrontendCancel
        217, -- FrontendSelect
        242, -- CursorScrollDown
        241, -- CursorScrollUp
        239, -- CursorX
        240, -- CursorY
        237, -- CursorAccept
        238, -- CursorCancel
        31,  -- MoveUpDown
        30,  -- MoveLeftRight
        21,  -- Sprint
        22,  -- Jump
        23,  -- Enter
        75,  -- VehicleExit
        71,  -- VehicleAccelerate
        72,  -- VehicleBrake
        59,  -- VehicleMoveLeftRight
        89,  -- VehicleFlyYawLeft
        9,   -- FlyLeftRight
        8,   -- FlyUpDown
        90,  -- VehicleFlyYawRight
        76,  -- VehicleHandbrake
        2,   -- LookUpDown
        1,   -- LookLeftRight
        25,  -- Aim
        24,  -- Attack
        71,  -- VehicleAccelerate
        72,  -- VehicleBrake
        59,  -- VehicleMoveLeftRight
        31,  -- MoveUpDown
        30,  -- MoveLeftRight
        75,  -- VehicleExit
    }
}, Controls)

---@comment Toggles all controls or only the ones used for the menu
---@param toggle boolean
function Controls:ToggleAll(toggle)
    if toggle then
        EnableAllControlActions(0)
        EnableAllControlActions(1)
        EnableAllControlActions(2)
    else
        DisableAllControlActions(2)

        local list = {}
        if IsUsingKeyboard(2) then
            list = self.NecessaryControlsKeyboard
        else
            list = self.NecessaryControlsGamePad
        end
        for _, control in pairs(list) do
            EnableControlAction(0, control, true)
        end
    end
end
